components/WorkSectionAsGallery.js

```diff
@@ -0,0 +1,133 @@
+import React from 'react';
+
+import PropTypes from 'prop-types';
+
+import { ContentfulMedia, SimpleFragment } from 'models';
+
+import { Markdown } from 'components/base';
+import Work from 'components/Work';
+
+import get from 'utils/get';
+
+const renderWorkGalleryAssetBlock = (workGalleryBlock) => {
+  const displayFirstAssetAsFullWidth = get(workGalleryBlock, 'fields.displayFirstAssetAsFullWidth', true);
+  const works = get(workGalleryBlock, 'fields.works', []);
+
+  return (
+    <div className="flex-col">
+      {displayFirstAssetAsFullWidth && 
+        <>
+          <div className="block col-8 relative">
+            <Work work={works[0]}/>
+          </div>
+          <div className="flex col-8 pb_5 md:pb1 items-end">
+            <div className="block col-4 pr_25 md:pr_5">
+              <div className="relative">
+                <Work work={works[1]}/>
+              </div>
+            </div>
+            <div className="block col-4 pr_25 md:pr_5">
+              <div className="relative">
+                <Work work={works[2]}/>
+              </div>
+            </div>
+          </div>
+        </>
+      }
+      {!displayFirstAssetAsFullWidth && 
+        <>
+          <div className="flex col-8 pb_5 md:pb1 items-end">
+            <div className="block col-4 pr_25 md:pr_5">
+              <div className="relative">
+                <Work work={works[0]}/>
+              </div>
+            </div>
+            <div className="block col-4 pr_25 md:pr_5">
+              <div className="relative">
+                <Work work={works[1]}/>
+              </div>
+            </div>
+          </div>
+          <div className="block col-8 relative">
+            <Work work={works[2]}/>
+          </div>
+        </>
+      }
+    </div>
+  );
+}
+
+const renderWorkGalleryTextBlock = (textBlock) => {
+  const text = get(textBlock, 'fields.text', '');
+  return (
+    <div>
+      {text &&
+        <Markdown
+          className="WorkSectionAsGallery__work-gallery-text my4 lg:my10 px1_25 md:px5 lg:px10 Markdown--medium"
+          src={text}
+        />
+      }
+    </div>
+  );
+};
+
+const WorkSectionAsGallery = (props) => {
+  const fields = get(props, 'workGallery.fields', {}); 
+  const workGalleryAssetBlocks = get(fields, 'workGalleryAssetBlocks', []);
+  const workGalleryTextBlocks = get(fields, 'workGalleryTextBlocks', []);
+  const renderAssetBlocksFirst = get(fields, 'renderAssetBlocksFirst', true);
+  const maximumLengthBlocks = workGalleryAssetBlocks.length > workGalleryTextBlocks.length ? [...Array(workGalleryAssetBlocks.length).keys()] : [...Array(workGalleryTextBlocks.length).keys()]; 
+  
+  return (
+    <div className="WorkSectionAsGallery mt2">
+      {maximumLengthBlocks.map((index) => {
+        const workGalleryBlock = renderWorkGalleryAssetBlock(workGalleryAssetBlocks[index]);
+        const textBlock = renderWorkGalleryTextBlock(workGalleryTextBlocks[index])
+        return (
+          <>
```

Comment:
we would want to use divs with keys here instead for the list items. https://reactjs.org/docs/lists-and-keys.html

Reply:
done :D